crates/host/src/context.rs

```diff
@@ -263,707 +263,510 @@ impl WasiHttpView for PluginCtx {
 
 impl crate::bindings::bulwark::plugin::types::Host for PluginCtx {}
 
+#[async_trait]
 impl crate::bindings::bulwark::plugin::config::Host for PluginCtx {
     /// Returns all config key names.
-    fn config_keys<'ctx, 'async_trait>(
-        &'ctx mut self,
-    ) -> Pin<Box<dyn Future<Output = Vec<String>> + Send + 'async_trait>>
-    where
-        'ctx: 'async_trait,
-        Self: 'async_trait,
-    {
-        Box::pin(async move { self.guest_config.keys().cloned().collect() })
+    async fn config_keys(&mut self) -> Vec<String> {
+        self.guest_config.keys().cloned().collect()
     }
 
     /// Returns the named config value.
-    fn config_var<'ctx, 'async_trait>(
-        &'ctx mut self,
+    async fn config_var(
+        &mut self,
         key: String,
-    ) -> Pin<
-        Box<
-            dyn Future<Output = Option<crate::bindings::bulwark::plugin::config::Value>>
-                + Send
-                + 'async_trait,
-        >,
-    >
-    where
-        'ctx: 'async_trait,
-        Self: 'async_trait,
-    {
-        Box::pin(async move {
-            let result = self
-                .guest_config
-                .get(key.as_str())
-                .map_or(Ok(None), |value| {
-                    // Invert, we need Result<Option<V>, E> rather than Option<Result<V, E>>.
-                    // This is also why the map_or default above is Ok(None).
-                    value.clone().try_into().map(Some)
-                });
-            result.expect("config should already be validated")
-        })
+    ) -> Option<crate::bindings::bulwark::plugin::config::Value> {
+        let result = self
+            .guest_config
+            .get(key.as_str())
+            .map_or(Ok(None), |value| {
+                // Invert, we need Result<Option<V>, E> rather than Option<Result<V, E>>.
+                // This is also why the map_or default above is Ok(None).
+                value.clone().try_into().map(Some)
+            });
+        result.expect("config should already be validated")
     }
 
     /// Returns the number of proxy hops expected exterior to Bulwark.
-    fn proxy_hops<'ctx, 'async_trait>(
-        &'ctx mut self,
-    ) -> Pin<Box<dyn Future<Output = u8> + Send + 'async_trait>>
-    where
-        'ctx: 'async_trait,
-        Self: 'async_trait,
-    {
-        Box::pin(async move { self.host_config.service.proxy_hops })
+    async fn proxy_hops(&mut self) -> u8 {
+        self.host_config.service.proxy_hops
```

Comment:
This analysis appears to be incorrect.

Reply:
The analysis is incorrect because the objective of the PR was not to make the functions `async` but rather to simplify their apparent interface with a macro.